### PR TITLE
Restructure user docs

### LINKED ISSUE #318: Mention support for nunjucks
Our documentation can mention that we support nunjucks out of the box. Furthermore, variables declared in markbind way can be accessed from nunjucks code globally.

### PR DESCRIPTION
**Pull request by damithc**
**What is the purpose of this pull request? (put "X" next to an item, remove the rest)**

• [x] Documentation update

Also fixes #318, #367, #371, #419

**What is the rationale for this request?**
Restructured and revised the documentation as its structure has deteriorated as a natural consequence of piece-meal updates over a long period.  I couldn't go as far as I wanted to due to lack of time; this version took the good part of a week :-p

Apologies for mutilating some of the content you wrote beyond recognition. In most cases, I started with the existing content but after numerous refactoring and revising, most are not traceable to the original content anymore. 

Also apologies for pushing as one commit. The actual commit history was too chaotic to make any sense.

Finally, apologies to those with ongoing PRs as most likely this PR is going to conflict with doc updates in your PR.

**Is there anything you'd like reviewers to focus on?**

Read from the perspective of a prospective user. Also look out for factual errors.

**Testing instructions:**

Use the netlify preview.

### CODE REVIEW

**Comment by Chng-Zhi-Xuan** [comment]
Thanks for the initial review @yamgent, will review in the next hour or so.

File: docs/index.md [hunk 1]
```diff
@@ -1,23 +1,40 @@
...
+<span class="lead">
 
-MarkBind is a website generator that can generate a website from markdown documents. While there are other markdown-to-html website generators around, **MarkBind can generate more dynamic websites**, as opposed to one-size-fits-all static content. MarkBind is particularly suitable to create content-heavy websites %%e.g., eLearning websites, online instruction manuals, project documentation etc.%% that aim for _<tooltip content="i.e., the reader can go deeper or get more content as desired">self-directed consumption</tooltip>_.   
+<big>**Generate dynamic websites from Markdown text.**</big>
 
-## <span class="glyphicon glyphicon-th-list" aria-hidden="true"></span> Features
+MarkBind can use Markdown documents to generate a <tooltip content="as opposed to _one-size-fits-all_ static content">_more dynamic_</tooltip> websites that facilitates _<tooltip content="i.e., the reader can chart their own path through the content and consume more/less content as desired">self-directed consumption</tooltip>_. MarkBind is ideal for creating content-heavy instructional websites %%e.g., eLearning websites, online instruction manuals, project documentation etc.%%.
```
Comment (Chng-Zhi-Xuan) [inline docs/index.md]: Add a `<br>` before 'e.g., eLearning .....' so the greyed text is all on 1 line.

**Comment by yamgent** [comment]
There are some recently approved PRs that has new addition to the documentation, and I think it would be better if those PRs actually adapt the new documentation format first before they get merged (rather than having it the other way round, which is this PR having to deal with the new additions).

We can merge this PR first even if there are some substantial flaws, and touch up in a subsequent PR?

**Comment by Chng-Zhi-Xuan** [comment]
@yamgent Good point, I was already thinking where to put page navigation in the refreshed docs as I was reviewing it 😅. Yes, we should prioritise merging this PR first.

**Comment by damithc** [comment]
> We can merge this PR first even if there are some substantial flaws, and touch up in a subsequent PR?

Yup, that makes sense. Anything I need to do?

**Comment by yamgent** [comment]
> Anything I need to do?

Not atm, the subsequent PR can be carried out by anyone available actually.

File: docs/common/header.md [hunk 2]
```diff
@@ -1,29 +1,12 @@
...
-    <li class="dropdown-header">Deploying a Site</li>
-    <li><a href="{{baseUrl}}/userGuide/ghpagesDeployment.html" class="dropdown-item">◦&nbsp; Github Pages Deployment</a></li>
-    <li><a href="{{baseUrl}}/userGuide/netlifyPreview.html" class="dropdown-item">◦&nbsp; Preview site on Netlify</a></li>
-    <li role="separator" class="dropdown-divider"></li>
-    <li><a href="{{baseUrl}}/userGuide/components.html" class="dropdown-item">Components Reference</a></li>
-  </dropdown>
+  <li><a href="{{baseUrl}}/index.html" class="nav-link">HOME</a></li>
+  <li><a href="{{baseUrl}}/userGuide/index.html" class="nav-link">DOCS</a></li>
+  <li><a href="{{baseUrl}}/showcase.html" class="nav-link">SHOWCASE</a></li>
+  <li><a href="{{baseUrl}}/about.html" class="nav-link">ABOUT</a></li>
```
Comment (acjh) [inline docs/common/header.md]: @damithc @yamgent It seems the removed dropdown means we can no longer navigate to those pages on mobile.

Comment (damithc) [reply docs/common/header.md]: That is indeed a problem. Ideally, the SiteNav should automatically transform into a drop-down menu. What do you guys think?

Comment (yamgent) [reply docs/common/header.md]: Created #591.

**Pull request merged**